website/cue/reference/remap/functions/map_keys.cue

```diff
@@ -0,0 +1,70 @@
+package metadata
+
+remap: functions: map_keys: {
+	category: "Enumerate"
+	description: #"""
+		Map the keys within an object.
+
+		If `recursive` is enabled, the function iterates into nested
+		objects, using the following rules:
+
+		1. Iteration starts at the root.
+		2. For every nested object type:
+		   - First return the key of the object type itself.
+		   - Then recurse into the object, and loop back to item (1)
+		     in this list.
+		   - Any mutation done on a nested object *before* recursing into
+		     it, are preserved.
+		3. For every nested array type:
+		   - First return the key of the array type itself
+		   - Then find all objects within the array, and apply item (2)
+		     to each individual object.
+
+		Practically speaking, the above rules mean that `map_keys` with
+		`recursive` enabled will find *all* keys in the target,
+		regardless of whether nested objects are nested inside arrays.
+
+		The function uses the "function closure syntax" to allow reading
+		the key for each item in the object.
+
+		The same scoping rules apply to closure blocks as they do for
+		regular blocks, meaning, any variable defined in parent scopes
+		are accessible, and mutations to those variables are preserved,
+		but any new variables instantiated in the closure block are
+		unavailable outside of the block.
+
+		Check out the examples below to learn about the closure syntax.
+		"""#
+
+	arguments: [
+		{
+			name:        "value"
+			description: "The object to iterate."
+			required:    true
+			type: ["object"]
+		},
+		{
+			name:        "recursive"
+			description: "Whether to recursively iterate the collection."
+			required:    false
```

Comment:
I think this needs a default defined?

Reply:
Resolved in [25852d9](https://github.com/vectordotdev/vector/pull/12721/commits/25852d9590d1e062eb7327c4dc3b3fbea6797b29).